src/components/Search/FreeSearchForm.tsx

```diff
@@ -1,132 +1,135 @@
+              ? resultList.map((r) => ({ ...listItemToOption(r), type: r.type }))
+              : [];
+
+            // This resultObject is needed to be able to add the type, value and name of the selected option,
+            // since the callback for selecting an option only returns the displayed value instead of the mapped object.
+            if (freeSearchFormItemList.length > 0) {
+              setSearchState((current) => ({
+                ...current,
+                resultObject: _.keyBy(freeSearchFormItemList, 'label'),
+              }));
             }
-            resolve([]);
+            resolve(freeSearchFormItemList || []);
           })
           .catch(() => reject([]));
       }),
-    [selections],
+    [],
   );
 
   /**
    * Gets the selectedValue and uses the resultObject to get the id and type for the
-   * selectedValue, if it exists in resultObject and isn't already added as a selection
+   * selectedValue, if it exists in resultObject and isn't already an added value
    */
   const handleSubmit = useCallback(
-    (value: string) => {
-      if (!arrayHasValue(selections, value) && objectHasProperty(resultObject, value)) {
-        dispatch(setFreeSearchParams({ [resultObject[value].label]: resultObject[value] }));
+    (value: string, onChange: (...event: unknown[]) => void) => {
+      const formValue = getValues('freeSearchParams');
+      if (!arrayHasValue(Object.keys(formValue), value) && objectHasProperty(resultObject, value)) {
+        const nextChange = {
+          ...formValue,
+          [resultObject[value].label]: resultObject[value],
+        };
+        onChange(nextChange);
         setSearchState((current) => {
           return {
             ...current,
             searchWord: '',
+            selections: Object.keys(nextChange),
           };
         });
       }
     },
-    [selections, resultObject, dispatch],
+    [resultObject],
   );
 
   /**
-   * Get the <span>-elements value and use the value to remove the key from redux
+   * Get the <span>-elements value and use the value to remove the key from the form value
    */
   const onSelectionDelete = useCallback(
-    (e: MouseEvent<HTMLButtonElement>) => {
-      dispatch(
-        removeFreeSearchParam(
-          (e.currentTarget as HTMLButtonElement)?.parentElement?.innerText as string,
-        ),
-      );
+    (e: MouseEvent<HTMLButtonElement>, onChange: (...event: unknown[]) => void) => {
+      const formValue = getValues('freeSearchParams') as FreeSearchFormObject;
+      // Copy everything except the chosen value to a new object, since react-hook-forms freezes the object
+      const {
+        [(e.currentTarget as HTMLButtonElement)?.parentElement?.innerText as string]: _,
+        ...nextChange
+      } = formValue;
+
+      onChange(nextChange);
+      setSearchState((current) => ({
+        ...current,
+        selections: Object.keys(nextChange || {}),
+      }));
     },
-    [dispatch],
+    [],
```

Comment:
add getValues as dependency, because it is coming from outside the component

Reply:
fixed